File: ete3/pairwise_patristic.py
#!/usr/bin/env python

from ete3 import Tree
import os
import sys
import time

# Get input
treefile=sys.argv[1]
print "reading",treefile
t = Tree( treefile )

# Start the clock
start = time.time()

# Run run run
leaves = t.get_leaves()
for n1 in xrange(len(leaves)):
    for n2 in xrange(n1+1, len(leaves)):
        t.get_distance(leaves[n1], leaves[n2])

# Stop the clock
end = time.time()
print("Internal time: ", end - start)

File: ape/pairwise_patristic.R
#!/usr/bin/env Rscript

library(ape)

# Get input tree file and read it (not part of this measurement)
args <- commandArgs(trailingOnly = TRUE)
treefile <- args[1]
print(paste("reading ",treefile))
tree <- read.tree(treefile)

# Start the clock
start <- Sys.time()

# Run, Forrest, Run!
mat <- cophenetic(tree)

# Stop the clock
end <- Sys.time()
duration <- difftime(end, start, units="secs")
print(paste("Internal time:", duration))

# Check the output
print(dim(mat))
#write.table( mat, "dist_mat.csv", row.names = FALSE, col.names = FALSE )
